db/encoding/string.go

```diff
@@ -0,0 +1,90 @@
+}
+
+// EncodeStringAscending encodes the string value using an escape-based encoding. See
+// EncodeBytes for details. The encoded bytes are append to the supplied buffer
+// and the resulting buffer is returned.
+func EncodeStringAscending(b []byte, s string) []byte {
+	return encodeStringAscendingWithTerminatorAndPrefix(b, s, ascendingBytesEscapes.escapedTerm, bytesMarker)
+}
+
+// encodeStringAscendingWithTerminatorAndPrefix encodes the string value using an escape-based encoding. See
+// EncodeBytes for details. The encoded bytes are append to the supplied buffer
+// and the resulting buffer is returned. We can also pass a terminator byte to be used with
+// JSON key encoding.
+func encodeStringAscendingWithTerminatorAndPrefix(
+	b []byte, s string, terminator byte, prefix byte,
+) []byte {
+	unsafeString := UnsafeConvertStringToBytes(s)
+	return encodeBytesAscendingWithTerminatorAndPrefix(b, unsafeString, terminator, prefix)
+}
+
+// EncodeStringDescending is the descending version of EncodeStringAscending.
+func EncodeStringDescending(b []byte, s string) []byte {
+	arg := UnsafeConvertStringToBytes(s)
+	return EncodeBytesDescending(b, arg)
+}
+
+// UnsafeConvertBytesToString performs an unsafe conversion from a []byte to a
+// string. The returned string will share the underlying memory with the
+// []byte which thus allows the string to be mutable through the []byte. We're
+// careful to use this method only in situations in which the []byte will not
+// be modified.
+func UnsafeConvertBytesToString(b []byte) string {
+	return *(*string)(unsafe.Pointer(&b))
+}
+
+// DecodeUnsafeStringAscending decodes a string value from the input buffer which was
+// encoded using EncodeString or EncodeBytes. The r []byte is used as a
+// temporary buffer in order to avoid memory allocations. The remainder of the
+// input buffer and the decoded string are returned. Note that the returned
+// string may share storage with the input buffer.
+func DecodeUnsafeStringAscending(b []byte, r []byte) ([]byte, string, error) {
+	b, r, err := DecodeBytesAscending(b, r)
+	return b, UnsafeConvertBytesToString(r), err
+}
+
+// DecodeUnsafeStringAscendingDeepCopy is the same as
+// DecodeUnsafeStringAscending but the returned string will never share storage
+// with the input buffer.
+func DecodeUnsafeStringAscendingDeepCopy(b []byte, r []byte) ([]byte, string, error) {
+	b, r, err := DecodeBytesAscendingDeepCopy(b, r)
+	return b, UnsafeConvertBytesToString(r), err
+}
+
+// DecodeUnsafeStringDescending decodes a string value from the input buffer which
+// was encoded using EncodeStringDescending or EncodeBytesDescending. The r
+// []byte is used as a temporary buffer in order to avoid memory
+// allocations. The remainder of the input buffer and the decoded string are
+// returned. Note that the returned string may share storage with the input
+// buffer.
+func DecodeUnsafeStringDescending(b []byte, r []byte) ([]byte, string, error) {
```

Comment:
todo: This appears to be dead code and only be called by unit tests, can you please remove it, along with  `DecodeUnsafeStringAscendingDeepCopy` which also appears to be dead.

Reply:
removed